test/kubernetes/integration_test.go

```diff
@@ -246,6 +246,138 @@ func TestDeployment_Single_Volume(t *testing.T) {
 	t.Log("Finished!")
 }
 
+func TestPersistentVolume_Resize(t *testing.T) {
+	volumeName := "my-do-volume"
+	claimName := "csi-pvc-resizer"
+
+	replicaCount := new(int32)
+	*replicaCount = 1
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-csi-app-resizer",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "my-csi-app",
+					Image: "busybox",
+					VolumeMounts: []v1.VolumeMount{
+						{
+							MountPath: "/data",
+							Name:      volumeName,
+						},
+					},
+					Command: []string{
+						"sleep",
+						"1000000",
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: volumeName,
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: claimName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	t.Log("Creating pod")
+	_, err := client.CoreV1().Pods(namespace).Create(pod)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pvc := &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: claimName,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				v1.ReadWriteOnce,
+			},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceStorage: resource.MustParse("5Gi"),
+				},
+			},
+			StorageClassName: strPtr(testStorageClass),
+		},
+	}
+
+	t.Log("Creating pvc")
+	_, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(pvc)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("Waiting pod %q to be running ...\n", pod.Name)
+	if err := waitForPod(client, pod.Name); err != nil {
+		t.Error(err)
+	}
+
+	createdPVC, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(claimName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// get pv name
+	pvName := createdPVC.Spec.VolumeName
+	pv, err := client.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if pv.Spec.Capacity["storage"] != resource.MustParse("5Gi") {
+		t.Fatal("initial volume size is not equal to requested volume size")
```

Comment:
Let's include the values we were comparing (`Spec.Capacity["storage"]` and `"5Gi"`) in the output.

Similar in the assertion below.